_posts/2024-02-19-jco-1.0.md

```diff
@@ -0,0 +1,301 @@
+---
+title: "Announcing Jco 1.0"
+author: "Yosh Wuyts"
+date: "2024-02-22"
+github_name: "yoshuawuyts"
+excerpt_separator: <!--end_excerpt-->
+---
+
+We’re happy to announce the 1.0 release of [Jco][jco]: a native Javascript
+WebAssembly toolchain and runtime built for WebAssembly Components and [WASI
+0.2](https://bytecodealliance.org/articles/WASI-0.2) [^wasi-one-liner]. Jco can
+natively run Wasm Components inside of Node.js, making it easy to take libraries
+written in different programming languages and execute them using the Node.js
+runtime. And by implementing the entirety of the WASI 0.2 API surface, those
+components can access the network, filesystem, and other system APIs available
+in the Node.js runtime.
+
+[^wasi-one-liner]: WebAssembly Components are an extension to WebAssembly to
+create strongly typed interfaces between different programs. WASI 0.2 is a
+re-work of the "WebAssembly System Interfaces" based on WebAssembly Components.
+The previous version of WASI was not based on components, and as a result was
+rather limited in what it could express.
+
+<!--end_excerpt-->
+
+Our goal is for Jco to be a comprehensive tool for all Component-related
+operations for JavaScript. With Jco 1.0 we're stabilizing a Node.js runtime for
+Wasm Components, as well as a toolchain to take Wasm Components written in other
+languages and import them into JavaScript. From here on out we hope to continue
+stabilizing more features for Jco. Some features are already available
+experimentally, including native support for the browser. As well as native
+support for compiling JavaScript code into WebAssembly.
+
+## JavaScript Toolchains in the Bytecode Alliance
+
+Jco is the third JS toolchain project which is part of the Bytecode Alliance. To
+help people navigate this, here is a quick overview of the three projects and
+what their purpose is at the time of writing:
+
+- [Javy](https://github.com/bytecodealliance/javy): A JavaScript → WASI 0.1
+compiler toolchain built using the [QuickJS JavaScript
+engine](https://bellard.org/quickjs/). This work predates Wasm Components, but
+support may be added in the future.
+- [ComponentizeJS](https://github.com/bytecodealliance/ComponentizeJS): A
+JavaScript → Wasm Component compiler toolchain with full support for WASI 0.2.
+This project is newer and not yet considered stable.
+- [Jco](https://github.com/bytecodealliance/jco): A comprehensive WebAssembly
+Component toolchain and runtime for JavaScript. This includes a Wasm Component → JS
+toolchain with full support for WASI 0.2 built using the [SpiderMonkey
+JavaScript engine](https://spidermonkey.dev).
```

Comment:
The second sentence here effectively repeats the componentizeJS definition, perhaps just say "also includes a ComponentizeJS wrapper" or something like that?

I know this is hard to communicate and we haven't made it easy to frame - but `jco componentize` is ComponentizeJS, perhaps there's a way to say that simply.

Reply:
I might just do away with this section since we're already mentioning it on line 27. This would be the fifth revision, and if we're still not getting it right then it's probably not worth keeping.

Reply:
I think this was posted on the wrong line? I suspect this was meant for line 52, which I've now removed (see earlier comment).

Reply:
Line 47 says "A JavaScript → Wasm Component compiler toolchain", then line 51 again says "This includes a Wasm Component → JS" toolchain.

If I didn't know any better I'd think jco is doing the same thing as ComponentizeJS? It's confusing.

Reply:
Latest looks good thanks. Some clarification around how componentize is made available in Jco could be useful, but it's certainly not necessary either.